### PR TITLE
Add entity type filtering in Token and Account kv states

### LINKED ISSUE #9862: Calls to HTS contracts are not properly propagated in reusable services
### Description

Specifically, contract calls that interact with HTS ERC functions like name, decimals, or totalSupply (which take a token address as a parameter) fail with CONTRACT_REVERT_EXECUTED.

For example, in the following method:

```
function name(address token) public view returns (string memory) {
    return IERC20Metadata(token).name();
}
```

When executing such calls, services tries to find the token in the method ->
`DispatchingEvmFrameState.getMutableAccount(Address address)`. 
This method searches for the token address in two stages:

Account Lookup:
Calls nativeOperations.getAccount(accountID) which checks AccountReadableKVState.
If the account is found, it is returned.

Token Lookup (if the account isn’t found):
Calls nativeOperations.getToken(number) which checks TokenReadableKVState.
If the token is found, it is returned.

The issue arises because AccountReadableKVState does not filter by entity type. If the token address is mistakenly treated as an account and returned from the account state, execution fails.

### Steps to reproduce

Explained in the description of the issue

### Additional context

_No response_

### Hedera network

other

### Version

v0.118.0

### Operating system

macOS

### PR DESCRIPTION
**Pull request by kselveliev**
**Description**:
This PR adds filtering for entity type in the account and token readable kv states.

This PR modifies ... in order to support ...
TokenReadableKVState - return token only if the entity type return from entity table is of type token
AccountReadableKVState - return the account when the entity type is not token
Unit test for the new cases

**Related issue(s)**:

Fixes #9862 

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/state/AccountReadableKVState.java [hunk 1]
```diff
@@ -92,6 +93,7 @@ protected Account readFromDataSource(@Nonnull AccountID key) {
         final var timestamp = ContractCallContext.get().getTimestamp();
         return commonEntityAccessor
                 .get(key, timestamp)
+                .filter(entity -> entity.getType() != TOKEN)
```
Comment (kselveliev) [inline hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/state/AccountReadableKVState.java:96]: Here if we use this filter `.filter(entity -> entity.getType() == ACCOUNT)` a couple of hundred tests start failing,
This is covering the case described in the issue that this pr fixes.

File: hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/state/AccountReadableKVStateTest.java [hunk 2]
```diff
@@ -163,6 +167,14 @@ void setup() {
         entity.setMaxAutomaticTokenAssociations(MAX_AUTOMATIC_TOKEN_ASSOCIATIONS);
         entity.setType(EntityType.ACCOUNT);
 
+        token = new Entity();
+        token.setId(TOKEN_NUM);
```
Comment (IvanKavaldzhiev) [inline hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/state/AccountReadableKVStateTest.java:171]: To be precise we should use EntityIdUtils to construct the ID based on SHARD, REALM, TOKEN_NUM and aggregate their values. The result will equal TOKEN_NUM but once we have shards and realms the values will differ.

File: hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/state/TokenReadableKVStateTest.java [hunk 3]
```diff
@@ -70,6 +70,8 @@ class TokenReadableKVStateTest {
 
     private static final TokenID TOKEN_ID =
             TokenID.newBuilder().shardNum(0L).realmNum(0L).tokenNum(1252L).build();
+    private static final TokenID TOKEN_ID_ACCOUNT =
```
Comment (IvanKavaldzhiev) [inline hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/state/TokenReadableKVStateTest.java:73]: We can just name it like TOKEN_ID_FOR_NEGATIVE_TEST or TOKEN_ID_SECOND or something like this.

**Review by steven-sheehy — DISMISSED** [review]

File: hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/state/AccountReadableKVStateTest.java (hunk 2 above)
Comment (kselveliev) [reply hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/state/AccountReadableKVStateTest.java]: Comment addressed

File: hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/state/TokenReadableKVStateTest.java (hunk 3 above)
Comment (kselveliev) [reply hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/state/TokenReadableKVStateTest.java]: Comment addressed

**Review by steven-sheehy — APPROVED** [review]

**Pull request merged**